File: MCCommandHelper-AIDE/app/src/main/java/com/mchelper/data/Command.java
package com.mchelper.data;

import java.util.List;
import java.util.ArrayList;

public class Command {
    private String id;
    private String name;
    private String syntax;
    private String description;
    private CommandCategory category;
    private List<String> examples;
    private String notes;

    public Command(String id, String name, String syntax, String description, 
                   CommandCategory category, List<String> examples, String notes) {
        this.id = id;
        this.name = name;
        this.syntax = syntax;
        this.description = description;
        this.category = category;
        this.examples = examples;
        this.notes = notes;
    }

    public String getId() { return id; }
    public String getName() { return name; }
    public String getSyntax() { return syntax; }
    public String getDescription() { return description; }
    public CommandCategory getCategory() { return category; }
    public List<String> getExamples() { return examples; }
    public String getNotes() { return notes; }
}

File: MCCommandHelper/app/src/main/java/com/mchelper/data/Command.kt
package com.mchelper.data

data class Command(
    val id: String,
    val name: String,
    val syntax: String,
    val description: String,
    val category: CommandCategory,
    val examples: List<String> = emptyList(),
    val notes: String = ""
)

enum class CommandCategory(val displayName: String, val icon: String) {
    GAMEPLAY("游戏玩法", "🎮"),
    WORLD("世界管理", "🌍"),
    PLAYER("玩家管理", "👤"),
    ENTITY("实体管理", "🐄"),
    ITEM("物品管理", "📦"),
    BLOCK("方块管理", "🧱"),
    SERVER("服务器", "🖥️"),
    CHEAT("作弊指令", "⚡"),
    REDSTONE("红石", "🔴"),
    OTHER("其他", "📋")
}
